### PR TITLE
op-node: reset back to optimal receipt fetching methods

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Reset it back after a while to not fallback permanently to sub-optimal RPC methods for receipt fetching.

**Tests**

Updated test to see if method preference can be sticky and reset back if configured to do so.

**TODOs**

- [x] Author or reviewer has added an entry to the [current release notes draft][RND], if appropriate.

[RND]: https://www.notion.so/oplabs/ded30107ceec41c88817e60322aa8d0a?v=b4a22cedb85a46a38c9be14e7c984953&pvs=4 "Release Notes"

Fix CLI-3697

### CODE REVIEW

File: op-node/sources/receipts_test.go [hunk 1]
```diff
@@ -226,6 +232,12 @@ func TestEthClient_FetchReceipts(t *testing.T) {
 			providerKind: RPCKindAlchemy,
 			setup:        fallbackCase(30, AlchemyGetTransactionReceipts),
 		},
+		{
+			name:         "alchemy sticky",
+			providerKind: RPCKindAlchemy,
+			staticMethod: true,
+			setup:        fallbackCase(30, AlchemyGetTransactionReceipts, AlchemyGetTransactionReceipts),
```
Comment (trianglesphere) [inline op-node/sources/receipts_test.go:239]: How is does this verify undoing the fallback?

**Review by trianglesphere — COMMENTED** [review]
Fallback seems to work. Do we know if we've hit this before?
Also, on the testing side, how does the test verify that we undo the fallback?

File: op-node/sources/receipts_test.go (hunk 1 above)
Comment (protolambda) [reply op-node/sources/receipts_test.go:239]: Compare it against the test case below, where it changes from AlchemyGetTransactionReceipts to EthGetBlockReceipts when the bool is not true.

**Review by trianglesphere — APPROVED** [review]
lgtm

**Pull request merged**